### ISSUE TITLE
[BUG]: Mojo install fails on Fedora 39

### ISSUE THREAD

**Issue by racecar-driver** [modularml/mojo#1816]
### Bug description

Running `modular install mojo` outputs:
```
[mojo][error] Mojo SDK post-install test "`mojo repl`" failed:
SubprocessError: Command '/home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo repl --one-line-before-file settings set target.load-cwd-lldbinit false --one-line-before-file settings set plugin.process.gdb-remote.packet-timeout 60' returned non-zero exit status 127.
  stderr:
    /home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo: /lib64/libtinfo.so.6: no version information available (required by /home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo)
    /home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo-lldb: error while loading shared libraries: libedit.so.2: cannot open shared object file: No such file or directory
  File "/home/adam/.modular/pkg/packages.modular.com_mojo/scripts/post-install/self-test.py", line 295, in test_mojo_repl
    raise SubprocessError(repl.returncode, " ".join(args), out, err)
[mojo][error] Some components of the Mojo SDK may have been installed successfully, but others may not work as expected.
[mojo][error] Please submit an issue to https://github.com/modularml/mojo and include the full output of the command you just ran.
modular: error: failed to run script
==========================
Failure Information:
	- Script: `PATH=/home/adam/.local/bin:/home/adam/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/home/adam/.dotnet/tools HOME=/home/adam PYTHONPATH=/home/adam/.modular/pkg/packages.modular.com_mojo/venv/lib/home/adam/.modular/pkg/packages.modular.com_mojo/venv/bin/python /home/adam/.modular/pkg/packages.modular.com_mojo/scripts/post-install/self-test.py --modular-home /home/adam/.modular --package-path /home/adam/.modular/pkg/packages.modular.com_mojo`
	- Result: 1
	- Stderr:
[mojo][error] Mojo SDK post-install test "`mojo repl`" failed:
SubprocessError: Command '/home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo repl --one-line-before-file settings set target.load-cwd-lldbinit false --one-line-before-file settings set plugin.process.gdb-remote.packet-timeout 60' returned non-zero exit status 127.
  stderr:
    /home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo: /lib64/libtinfo.so.6: no version information available (required by /home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo)
    /home/adam/.modular/pkg/packages.modular.com_mojo/bin/mojo-lldb: error while loading shared libraries: libedit.so.2: cannot open shared object file: No such file or directory
  File "/home/adam/.modular/pkg/packages.modular.com_mojo/scripts/post-install/self-test.py", line 295, in test_mojo_repl
    raise SubprocessError(repl.returncode, " ".join(args), out, err)
[mojo][error] Some components of the Mojo SDK may have been installed successfully, but others may not work as expected.
[mojo][error] Please submit an issue to https://github.com/modularml/mojo and include the full output of the command you just ran.


==========================

```

### Steps to reproduce

Following standard setup procedure on a Fedora 39 system.  Output from `modular install mojo` above.

### System information

```shell
- What OS did you do install Mojo on ?
Fedora 39
- Provide version information for Mojo by pasting the output of `mojo -v`
N/A - mojo install fails.
- Provide Modular CLI version by pasting the output of `modular -v`
modular 0.5.0 (47924ae4)
```

**Comment by Sharktheone** [comment]
For the `libtinfo` issue, see #855

TL;DR use [my script](https://github.com/Sharktheone/arch-mojo) for the installation on fedora or arch
BTW it needs at least Python 3.10

**Comment by TrevorBenson** [comment]
> For the `libtinfo` issue, see #855
> 
> TL;DR use [my script](https://github.com/Sharktheone/arch-mojo) for the installation on fedora or arch BTW it needs at least Python 3.10

In #855 I only see a reference to making a libedit symlink, not libtinfo. I also reviewed your script, which copies a Debian `libtinfo.so.6.4`. Since this library is installed during initial OS installation and is a dependency I was hesitant to test replacing it since the RPM that provides it, **ncurses-libs** also cannot be uninstalled due to dependency chains. 

So, I created an F39 VM and tested this out, as soon as the libtinfo file is overwritten it starts to cause core dumps until the terminal finally crashes. New shells continue to cause additional core dumps until every existing terminal window is crashed/closed which eventually leads to the entire UI/desktop crashing and forces the user to log out. 

If I was more aware of Debian builds and where to check for the source of that deb package I would attempt to isolate the difference between the two `libtinfo.so.6.4` libraries, as they have different md5sums. Then open a ticket w/ Fedora to include whatever version details seem to be missing.

I am hesitant to use this script as a Fedora solution. During future OS upgrades, it could core dump again, and potentially with extremely bad timing during the upgrade leading to further unexpected issues. It would also need to be re-applied after updates, which could easily lead to additional core dumps, etc..

**Comment by Sharktheone** [comment]
> So, I created an F39 VM and tested this out, as soon as the libtinfo file is overwritten it starts to cause core dumps until the terminal finally crashes. Trying to open a new terminal shell causes additional core dumps, as does accessing existing terminal windows. Until every one of them is crashed/closed, which eventually leads to the entire UI/desktop crashing and forces the user to log out.

That's interesting. I wasn't aware of that. Normally my script shouldn't override anything. It only places it's libraries to a own directory in `.local` if you aren't installing it globally. The other option to fix this problem is to put the libs directly in the mojo installation directory, however this can cause problems with `modular clean`.

Have you used the script in global or local mode? If it was in global mode I might just disable it for the fedora installation.

**Comment by TrevorBenson** [comment]
Here is a breakdown of the script as a solution for Fedora. I ran all updates this after install of F39, last time I did not update to the latest packages on an existing F39 VM:
1. Using `--fedora` places files in `~/.local/lib/mojo/`, has no effect on fixing the error: `/lib64/libtinfo.so.6: no version information available` during `modular install mojo`.
2. Using `--global --fedora` places the lib in `/usr/lib/` and a symlink. Does not place it inside `/usr/lib64`, the actual path where it is expected and being found even after sourcing `~/.bashrc`. Has no effect on fixing the error  `/lib64/libtinfo.so.6: no version information available` during `modular install mojo`.
3. After hitting this point yesterday I tested:
   * Swapping the order of LD_LIBRARY_PATH to `export LD_LIBRARY_PATH=~/.local/lib/mojo:$LD_LIBRARY_PATH`, sourcing `.bashrc` again, and attempting to install. Also had no impact on removing the errors about libtinfo.so.6 version information.
   * Since no other methods resolved the issue I copied the library into `/usr/lib64` to replace the existing library. This is when core dumps started, I didn't see if replacing the symlink only made any difference, as I suspect everything should be referencing the so.6 version, not the explicit so.6.4.
     * However, after the full crash of the UI, logging out and back in, the error about version information is gone. The `libedit.so.2` error still occurred, but that is because crashes start before being able to symlink it, and after logging back in I decided to test the libtinfo fix by itself, which did resolve the error.

That's the point where I posted a response yesterday and decided to skip this method as a stable workaround to the issue for Fedora. I could use this solution inside a VM. However, I'd rather run Mojo on my actual development system instead of maintaining a dedicated system. If Fedora support isn't going to happen any time soon I'll reconsider a vscode remote host or maintaining a special devcontainer, but only as a last resort.
